INSTALL.rst

```diff
@@ -16,7 +16,17 @@ Matplotlib and its dependencies are available as wheel packages for macOS,
 Windows and Linux distributions::
 
   python -m pip install -U pip
-  python -m pip install -U matplotlib
+  python -m pip install --no-user --prefer-binary -U matplotlib
+
+The ``--prefer-binary`` flag will select the newest version of
+Matplotlib for which there is a wheel.  Remove this flag if you are
+willing to possibly compile Matplotlib from source.
+
+As of pip 20.1, if the :file:`site-packages` folder is not writable,
+it will fallback to behaving as if ``--user`` was passed.  In some
+cases this behavior is desirable, it can also lead to very confusing
+installation issues.  If you are comfortable with that risk, then
+remove ``--no-user``.
```

Comment:
This is basically saying "we don't trust pip devs to have made the right choice with fallback-to-user-by-default so we'll overrule them", so I don't really like it tbh.

Reply:
hmm, I think I have been conflating `--user` and `--home` in my head.  I was under the incorrect impression that `--user` did not put the python version in the path it installs (`--user` does [1] for the modules and headers, but not scripts) so my biggest concern (that you will get compiled extensions for the wrong version of python) is unfounded.

I've become convinced I am wrong on the merits (but I still don't like `--user`).

[1] https://docs.python.org/3.8/install/index.html#alternate-installation-the-user-scheme